### ISSUE TITLE
Star Wars force unleashed [Screen Overlay problem]

### ISSUE THREAD

**Issue by Levan7** [hrydgard/ppsspp#9572]
I am using the PPSSPP v1.4-56-gb1a530b Windows 64 bit build and started Forced unleashed and noticed that the game has a screen overlay in game, the menus work fine.

[DebugLog.zip](https://github.com/hrydgard/ppsspp/files/907484/DebugLog.zip)
![ulus10345_00000](https://cloud.githubusercontent.com/assets/5770076/24829454/f91a8e6c-1c82-11e7-9a5c-3af540969177.jpg)

**Comment by sum2012** [comment]
Does v1.3 have his problem ?

**Comment by Levan7** [comment]
@sum2012 
Hm, no it doesn't strange
![ulus10345_00000](https://cloud.githubusercontent.com/assets/5770076/24829758/f83e4780-1c88-11e7-8078-f3e5c76003c6.jpg)

**Comment by sum2012** [comment]
Can you test v1.3-xxx to find out which version first have problem in
http://buildbot.orphis.net/ppsspp/index.php?m=fulllist ?

**Comment by Levan7** [comment]
@sum2012 
Ok i found a problem with my report, when i was regressive testing. I noticed that sometimes the 1.3.xx builds get that over layer and sometime they don't (default settings) even the  PPSSPP v1.4-56-gb1a530b didn't get that over layer after few tries 
![ulus10345_00000](https://cloud.githubusercontent.com/assets/5770076/24829935/dd9b6210-1c8c-11e7-86f9-bd32bf06f962.jpg)

**Comment by hrydgard** [comment]
Try this:  Right click on the game to get the game ID (4 letters, 5 numbers). 

Then open assets/compat.ini and enter the game ID under [ClearToRAM] like the others.

Does that help?

**Comment by Levan7** [comment]
@hrydgard 
Well it fills like it did help, at least couldn't reproduce after trying multiple times.
 i made a entry
under [ClearToRAM] 
# Star Wars: The Force Unleashed
ULUS10345 = true

**Comment by unknownbrackets** [comment]
Possibly related: #3064

-[Unknown]

**Comment by Levan7** [comment]
Still an issue on v1.5.4-376-g3aca9e3ca

**Comment by Jinzombie** [comment]
Still have an issue on the latest ppsspp version 1.6.2
but restarting the ppsspp fix the issue.

**Comment by unknownbrackets** [comment]
Does this only happen when using frameskip, fast forward, or alternate speed?  Or does it happen even if you don't use those features?

-[Unknown]

**Comment by Jinzombie** [comment]
Default Settings this screen overlay still happen and only happen in the 1st boot/start of the game after that I changed from buffered rendering to skip-buffer effects and restart the ppsspp it's fix the problem

**Comment by Jinzombie** [comment]
and I think 4XMSAA also help to fix this issue since I always enable that in the developers options in Android Settings 😅

**Comment by hrydgard** [comment]
How about now?

**Comment by jullebarge** [comment]
I have the same problem on Android with version 1.9.3-164.
Tried various settings but nothing remove those lines.

**Comment by hrydgard** [comment]
Don't fast-forward on the loading screens and it should be OK. Use Rendering Mode = Buffered.

**Comment by jullebarge** [comment]
I don't use fast forward and I'm already in buffered mode, still I have those red lines.

**Comment by 2bndy5** [comment]
Posted this in the closed (seemingly identical) issue by mistake. https://github.com/hrydgard/ppsspp/issues/3064#issuecomment-622278383

**Comment by Gamemulatorer** [comment]
@Levan7 @jullebarge @2bndy5 Could you create a frame dump.? Doesn't matter if the frame dump is created on PC or Android.

https://github.com/hrydgard/ppsspp/wiki/How-to-create-a-frame-dump

**Comment by 2bndy5** [comment]
[with red screen - force unleashed - ppsspp debug.zip](https://github.com/hrydgard/ppsspp/files/4727844/with.red.screen.-.force.unleashed.-.ppsspp.debug.zip)

**Comment by Gamemulatorer** [comment]
@2bndy5 thanks for the dump
Here's the results (OpenGL)
### Hardware Rendering
![Screenshot_20200604-141914HW](https://user-images.githubusercontent.com/37603562/83722019-d053fd00-a66e-11ea-9acf-01f22d63792b.png)

### Software Rendering
![Screenshot_20200604-142028SW](https://user-images.githubusercontent.com/37603562/83722036-d77b0b00-a66e-11ea-847e-70e047868011.png)

**Comment by Kzitold** [comment]
I had to switch the backend to Direct3D 9.

**Comment by K0bin** [comment]
I can reproduce this on my phone (Adreno 650, driver version: 512.501.0) but not on my PC (GTX 1070, driver version: 461,40) both using the Vulkan renderer. So this might be a driver bug.

**Comment by K0bin** [comment]
This seems to be fixed in 1.11 bit there are other problems now.

The scene seems a bit too dark, fog is too strong on character models and the lightsaber trail effect has a weird checkerboard pattern.

![Screenshot_20210209-120240_2f85358b2198d26f8aca533d68bee793.jpg](https://user-images.githubusercontent.com/1131720/107355134-5b737580-6acf-11eb-987b-778ec23af2e2.jpg)

![Screenshot_20210209-120203_2f85358b2198d26f8aca533d68bee793.jpg](https://user-images.githubusercontent.com/1131720/107355156-629a8380-6acf-11eb-8bc8-c95a7a044997.jpg)

**Comment by sum2012** [comment]
We have applied Adreno driver fix

**Comment by K0bin** [comment]
What do you mean, is there a workaround for a driver bug in PPSSPP now? If so, can you link the commit, I'm curious.

**Comment by sum2012** [comment]
@K0bin #13910

**Comment by unknownbrackets** [comment]
Could you try exporting a GE frame dump of the lightsaber problem?  These help a lot.  Just make sure to export it when you see the issue - it's almost like a screenshot.  If you make it while the issue isn't happening, it won't help.

See here for instructions - it's not hard and works on Android too:
https://github.com/hrydgard/ppsspp/wiki/How-to-create-a-frame-dump

You can zip that and then drag and drop it into a reply here.

-[Unknown]

**Comment by K0bin** [comment]
It only happens when hardware transform is turned off, so I'm not sure if this is actually a regression.
Also noteworthy that I can reproduce it on Windows (Nvidia, Vulkan renderer).
[ULES00982_0009.zip](https://github.com/hrydgard/ppsspp/files/5953365/ULES00982_0009.zip)

The fog/brightness bug does not happen on Windows with the Vulkan renderer. It does happen with D3D11 though.

And I also managed to get the overlay again. This time on Windows with Vulkan.
![grafik](https://user-images.githubusercontent.com/1131720/107408923-e4a89d80-6b0b-11eb-8ddd-d049eb91ddab.png)

**Comment by Gamemulatorer** [comment]
I can't reproduce this using the dump both OGL and VK backend.
![Screenshot_2021-05-02-22-30-21-931_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/116816808-132b9700-ab96-11eb-8b1c-ea2bebb30681.jpg)
![Screenshot_2021-05-02-22-30-01-626_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/116816812-14f55a80-ab96-11eb-849f-5f1c03c7d637.jpg)

**Comment by Gamemulatorer** [comment]
I tried this game the red screen overlay still present on both OpenGL and Vulkan, but in OpenGL this game is heavily lag unlike Vulkan this game is running full speed.

### OpenGL
![Screenshot_2021-05-23-11-04-56-707_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/119246798-12938880-bbb7-11eb-9601-881904df9445.jpg)
### Vulkan
![Screenshot_2021-05-23-11-05-51-016_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/119246802-1e7f4a80-bbb7-11eb-956c-b477a080fecd.jpg)

**Comment by Levan7** [comment]
I tasted this and noticed that it happened on PC Vulkan when fast forwarding in loading maybe the emulation has problems when loading faster or slower than it should ?

**Comment by Gamemulatorer** [comment]
@Levan7 try this cheat code
_C1 Red Overlay Fix
_L 0xE001ADAE 0x003E94BE
_L 0x203E94BC 0xADA00000
I tried that cheat code both OpenGL and Vulkan and result is good 👍
![Screenshot_2021-05-23-20-04-34-065_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/119259796-8dce5c00-bc02-11eb-824c-5fc2b99c8524.jpg)

**Comment by Panderner** [comment]
Vulkan performance for my POCO M3 is slower than OpenGL in 4x resolution

OpenGL:
![Screenshot_2021-05-31-16-34-54-408_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/120166010-bff94280-c22e-11eb-88d9-a05d4a57efb9.jpg)

Vulkan:
![Screenshot_2021-05-31-16-35-43-721_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/120166062-cbe50480-c22e-11eb-8366-966174ae2488.jpg)

And I'm not using red overlay cheat.

**Comment by hrydgard** [comment]
good to know, thanks!

**Comment by Gamemulatorer** [comment]
> Vulkan performance for my POCO M3 is slower than OpenGL in 4x resolution
> 
> OpenGL:
> ![Screenshot_2021-05-31-16-34-54-408_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/120166010-bff94280-c22e-11eb-88d9-a05d4a57efb9.jpg)
> 
> Vulkan:
> ![Screenshot_2021-05-31-16-35-43-721_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/120166062-cbe50480-c22e-11eb-8366-966174ae2488.jpg)
> 
> And I'm not using red overlay cheat.

With red overlay cheat code this game is running full speed at 4X rendering resolution on my RedmiNote 9

### OPENGL
![Screenshot_2021-05-31-19-03-16-514_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/120184000-0bb5e700-c243-11eb-9268-0d43e288cda5.jpg)
### VULKAN
![Screenshot_2021-05-31-19-02-41-957_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/120184047-18d2d600-c243-11eb-9ccc-31b64cd74dba.jpg)
The red overlay is the cause of lag on my phone 🤷

**Comment by Gamemulatorer** [comment]
@Panderner I also notice on your screenshots that is too dark compare to mine 🤣

**Comment by Panderner** [comment]
I tested Windows 10 PC The Brightness is ok for OpenGL and Vulkan but it's too dark for D3D9 and D3D11
![2021-05-31 (4)](https://user-images.githubusercontent.com/37503397/120190878-c5b15100-c24b-11eb-96a6-bff7b2c5b5b7.png)
![2021-05-31 (5)](https://user-images.githubusercontent.com/37503397/120190897-c944d800-c24b-11eb-96d8-bb2c8618ea99.png)

**Comment by Panderner** [comment]
@Gamemulatorer Red Overlay Fix cheat can cause graphical glitches and FMV videos does not display

Here's a Title Screen:
![Screenshot_2021-05-31-20-03-31-691_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/120191186-33f61380-c24c-11eb-8b84-96124f3b9a33.jpg)

**Comment by Gamemulatorer** [comment]
Adding this game to PixelDepthRounding compat.ini sometimes fix this red overlay screen issue on my phone.
### OpenGL
![Screenshot_2021-05-31-22-39-07-508_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/120209779-a7eee680-c261-11eb-901a-239449670980.jpg)
### Vulkan
![Screenshot_2021-05-31-22-38-24-814_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/120209825-b3421200-c261-11eb-8445-40f4d52ca81f.jpg)
Redmi Note 9 Mediatek Helio G85 Mali G52 MC2 GPU

**Comment by hrydgard** [comment]
Only sometimes?

I don't know how that setting could be related though..

**Comment by Gamemulatorer** [comment]
> Only sometimes?
> 
> I don't know how that setting could be related though..

On OpenGL 5 out of 10 tries the screen overlay will show up but on Vulkan only twice out of 10 tries the red overlay screen show up.

**Comment by MojoJojoDojo** [comment]
This still occuring issue is annoying.

**Comment by ResurrectionRedmi9** [comment]
This issue is very rare to happen using the latest git but yeah still annoying issue.

**Comment by Gamemulatorer** [comment]
> This still occuring issue is annoying.

@MojoJojoDojo Try my work around if your playing this game on pc see ~ https://github.com/hrydgard/ppsspp/issues/12949#issuecomment-934251037

**Comment by Gamemulatorer** [comment]
Changing I/O method doesn't really fix this the only solution I found is by adding this game to Force04154000Download compatibility.

**Comment by CristobalManga** [comment]
> Cambiar el método de E / S realmente no soluciona esto, la única solución que encontré fue agregar este juego a la compatibilidad Force04154000Download.

In fact, if it fixes the problem.
But this way.
When the problem is active.
When you change the I / O, the error is fixed.
No matter which option you choose.
By the way, can you tell me where do I get compat.ini, in android? On PC I get it easy, but on android I can't get it.

**Comment by Gamemulatorer** [comment]
Here's the proof!
⬇️

**Force04154000Download** #OFF

https://user-images.githubusercontent.com/37603562/138015796-71e74409-2527-4e6f-8e86-919c9f935324.mp4

https://user-images.githubusercontent.com/37603562/138015810-9980ca40-a189-4b24-8724-f7bb680ea8a1.mp4

**Force04154000Download** #ON

https://user-images.githubusercontent.com/37603562/138015846-31f1b0d4-193a-4609-8c0f-8eddd25e36bd.mp4

https://user-images.githubusercontent.com/37603562/138015858-a2700153-1ee8-468b-a97f-fac8883434aa.mp4

Here's my compat.ini file @CristobalManga 
⬇️
[compatibility.zip](https://github.com/hrydgard/ppsspp/files/7377652/compatibility.zip)

**Comment by 2bndy5** [comment]
@Gamemulatorer Your compat.ini does fix the graphics issue, but it makes the game speed unplayable. Please stop spamming this fix. I'd rather see this issue resolved in a more sustainable way.

@CristobalManga I think the compay.ini file needs to be created in the _PSP/SYSTEM_ folder on your android devices internal storage. The directory should already exist once the PPSSPP emulator is installed. However, here is the warning at the top of the compat.ini file:

> This file is not meant to be user-editable, although is kept as a separate ini
> file instead of compiled into the code for debugging purposes.
>
> The uses cases are strict:
>   * Enable fixes for things we can't reasonably emulate without completely ruining
>     performance for other games, such as the screen copies in Dangan Ronpa
>   * Disabling accuracy features like 16-bit depth rounding, when we can't seem to
>     implement them at all in a 100% compatible way
>   * Emergency game-specific compatibility fixes before releases, such as the GTA
>     music problem where every attempted fix has reduced compatibility with other games
>   * Enable "unsafe" performance optimizations that some games can tolerate and
>     others cannot. We do not currently have any of those.
>
> This functionality should NOT be used for any of the following:
>   * Cheats
>   * Fun hacks, like enlarged heads or whatever
>   * Fixing general compatibility issues. First try to find a general solution. Try hard.

TL;DR -> compat.ini is not a viable solution
@unknownbrackets @MojoJojoDojo I look forward to your investigations.

**Comment by Gamemulatorer** [comment]
> @Gamemulatorer Your compat.ini does fix the graphics issue, but it makes the game speed unplayable. Please stop spamming this fix. I'd rather see this issue resolved in a more sustainable way.

Im just suggesting a workaround lol 😂
I also mentioned about the performance penalty in https://github.com/hrydgard/ppsspp/issues/12949#issuecomment-934251037

**Comment by Gamemulatorer** [comment]
This issue can be fixed by adding this game to **AllowLargeFBTextureOffset** see ~ https://github.com/hrydgard/ppsspp/pull/15691

**Comment by SilentMRG** [comment]
Hello...

Unfortunately, the bug is still present, but this time it's not red, but transparent, it even looks like a CRT filter, with vertical lines, hehehe.

Strangely, the bug occurs in the first Starkiller mission and the third mission in Raxus Prime. The Darth Vader quest and the Jedi Temple quest do not have the bug. And I didn't play the game to the end so I can't say if the bug recurs in other stages.

All the rest of the effects work like a charm! The frame rate is also very good and I haven't seen any other bugs.

The game has been tested in DirectX 9 and OpenGL. Not tested in DirectX 11 as my Laptop (Windows 7 Ultimate 64-bit - i5 2.64 GHz - 4GB DDR3 RAM and Intel HD Graphics 1000) is very old and doesn't support this API.

Thank you for everything!

**Comment by SilentMRG** [comment]
Hello...

Unfortunately, the bug is still present, but this time it's not red, but transparent, it even looks like a CRT filter, with vertical lines, hehehe.

Strangely, the bug occurs in the first Starkiller mission and the third mission in Raxus Prime. The Darth Vader quest and the Jedi Temple quest do not have the bug. And I didn't play the game to the end so I can't say if the bug recurs in other stages.

All the rest of the effects work like a charm! The frame rate is also very good and I haven't seen any other bugs.

The game has been tested in DirectX 9 and OpenGL. Not tested in DirectX 11 as my Laptop (Windows 7 Ultimate 64-bit - i5 2.64 GHz - 4GB DDR3 RAM and Intel HD Graphics 1000) is very old and doesn't support this API.

Thank you for everything!

**Comment by Gamemulatorer** [comment]
@SilentMRG use hack setting **lower resolution for effects (reduce artifacts)** to fix the crt line graphic glitch.

**Comment by 2bndy5** [comment]
I would consider the pseudo CRT filter as a new issue. Even if it a regression, it should get its own thread/issue.

**Comment by SilentMRG** [comment]
> use hack setting **lower resolution for effects (reduce artifacts)** set to aggresive to fix the crt line graphic glitch.

It worked my man! Thank you very much! =D You saved my hours of fun. \o/
